src/_pytest/config/__init__.py

```diff
@@ -732,7 +738,9 @@ class InvocationParams:
         plugins = attr.ib()
         dir = attr.ib(type=Path)
 
-    def __init__(self, pluginmanager, *, invocation_params=None) -> None:
+    def __init__(
+        self, pluginmanager: "PytestPluginManager", *, invocation_params=None
```

Comment:
Maybe `invocation_params: Optional[Config.InvocationParams]`?

(OK if you left it out intentionally).